test/extensions/stats_sinks/metrics_service/grpc_metrics_service_impl_test.cc

```diff
@@ -298,24 +305,86 @@ TEST_F(MetricsServiceSinkTest, FlushPredicate) {
 
   // Default predicate only accepts used metrics.
   {
-    MetricsFlusher flusher(true, true);
+    MetricsFlusher flusher(true, true,
+                           envoy::config::metrics::v3::HistogramEmitMode::SUMMARY_AND_HISTOGRAM);
     auto metrics = flusher.flush(snapshot_);
     EXPECT_EQ(1, metrics->size());
   }
 
   // Using a predicate that accepts all metrics, we'd flush both metrics.
   {
-    MetricsFlusher flusher(true, true, [](const auto&) { return true; });
+    MetricsFlusher flusher(true, true,
+                           envoy::config::metrics::v3::HistogramEmitMode::SUMMARY_AND_HISTOGRAM,
+                           [](const auto&) { return true; });
     auto metrics = flusher.flush(snapshot_);
     EXPECT_EQ(2, metrics->size());
   }
 
   // Using a predicate that rejects all metrics, we'd flush no metrics.
-  MetricsFlusher flusher(true, true, [](const auto&) { return false; });
+  MetricsFlusher flusher(true, true,
+                         envoy::config::metrics::v3::HistogramEmitMode::SUMMARY_AND_HISTOGRAM,
+                         [](const auto&) { return false; });
   auto metrics = flusher.flush(snapshot_);
   EXPECT_EQ(0, metrics->size());
 }
 
+// Test that verifies counters are reported as the delta between flushes when configured to do so.
+TEST_F(MetricsServiceSinkTest, HistogramEmitMode) {
+  addHistogramToSnapshot("test_histogram");
+
+  {
+    // This test will emit summary and histogram.
+    MetricsServiceSink<envoy::service::metrics::v3::StreamMetricsMessage,
+                       envoy::service::metrics::v3::StreamMetricsResponse>
+        sink(streamer_, true, false,
+             envoy::config::metrics::v3::HistogramEmitMode::SUMMARY_AND_HISTOGRAM);
+
+    EXPECT_CALL(*streamer_, send(_)).WillOnce(Invoke([](MetricsPtr&& metrics) {
+      ASSERT_EQ(2, metrics->size());
+      EXPECT_EQ("test_histogram", (*metrics)[0].name());
+      EXPECT_EQ("test_histogram", (*metrics)[1].name());
+
+      const auto& metric1 = (*metrics)[0].metric(0);
+      ASSERT(metric1.has_summary());
```

Comment:
s/ASSERT(/EXPECT_TRUE(/ here and below.

ASSERT will crash the test binary if it fails. ASSERT_TRUE will fail the test and early-exit the method if it fails,  and EXPECT_TRUE will fail the test and continue. I think EXPECT_TRUE is what you want here.